CRVReco/src/CrvCoincidenceFinder_module.cc

```diff
@@ -443,8 +454,24 @@ namespace mu2e
       //don't store clusters that are below the minimum number of PEs for this sector (or sectors, if the cluster involves multiple sectors)
       if(PEs<minClusterPEs) continue;
 
+      //find average hit times on both sides of the modules
+      std::array<size_t, CRVId::nSidesPerBar> sideHits{recoTimes[0].size(),recoTimes[1].size()};
+      std::array<float, CRVId::nSidesPerBar>  sidePEs{0,0};
+      std::array<double, CRVId::nSidesPerBar> sideTimes{0,0};
+      for(size_t side=0; side<CRVId::nSidesPerBar; ++side)
+      {
+        for(auto recoTimeIter=recoTimes[side].begin(); recoTimeIter!=recoTimes[side].end(); ++recoTimeIter)
+        {
+          //use fit values and not the no-fit option
+          sidePEs[side]+=recoTimeIter->second->_crvRecoPulse->GetPEs();
+          sideTimes[side]+=recoTimeIter->second->_crvRecoPulse->GetPulseTime()*recoTimeIter->second->_crvRecoPulse->GetPEs();  //PE-weighted pulse time average
+        }
+        if(sidePEs[side]>0) sideTimes[side]/=sidePEs[side]; else sideTimes[side]=0.0;
```

Comment:
@kutschke @brownd1978 Can invalid times be set to -inf? Or could it be set to -1? I don't know, if we can have negative times. Or should I add an additional invalid flag? Currently, invalid times are indicated by PEs being 0 and/or number of hits being 0.

Reply:
It really depends on downstream usage. Just checking one thing - are you talking about flagging a CrvCoincidenceCluster as invalid or some other type of object?     Can you explain why you want to keep invalid objects?    It sounds to me like #PE=0  is a sufficient validity flag.  You can improve the class by adding a member function like

`bool isValid() const { return _PEs != 0.;}` 

About downstream usage, is a time of zero valid if there are non-zero PEs?  If so, do we have time cuts that exclude for objects of physics interest?

Reply:
The coincidence cluster would still be valid, but one of the "side times" may not be valid, e.g. because a module has only one readout side, or there weren't sufficient hits to form a coincidence on of the sides. If this happens, then the PEs of this readout side is 0. So PEs[side]==0 would be the check for validity. If the PEs[side] are not zero, then the times will always be valid.

Reply:
I think you are saying that most CoincidenceClusters have two valid sides but some only have one.  There will never be one with no valid sides.  Is that right?

I think we still have not converged on the value of time for an invalid side.  My preference is this: if there will never be a valid time of zero, then set invalid times to zero.  If there can be valid times of zero, then set it to -1.    I don't like -inf.

Instead of isValid, I suggest a member function:

  `bool bothSidesValid() const {  return (_sidePEs[0] > 0. && _sidePEs[1] > 0);}`